experimental/PropertyDDS/packages/property-inspector-table/src/utils.tsx

```diff
@@ -630,8 +630,8 @@ export const collectionChildTableRow = (collectionPropertyProxy: BaseProxifiedPr
         property = (referencedPropertyParent as ContainerProperty).get(relativePathFromParent)!;
```

Comment:
ContainerProperty.get(..) should return by default `undefined | BaseProperty`, so type casting below shouldn't be necessary, its not clear to me what changed in TS version 4.5.5 that it doesn't resolve the type correctly anymore. 

https://github.com/microsoft/FluidFramework/blob/7bf8db9bc08f3d3f12a1cffa585a552785cbcf2b/experimental/PropertyDDS/packages/property-properties/src/index.d.ts#L573

Reply:
In typescript 4.5.5, the inferred type at that location is `BaseProperty | BaseProxifiedProperty<BaseProperty>`. This appears to be a regression in type inference in this specific case. I have pushed a workaround.